glusterd2/commands/snapshot/snapshot-restore.go

```diff
@@ -0,0 +1,375 @@
+package snapshotcommands
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/snapshot"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	volgen "github.com/gluster/glusterd2/glusterd2/volgen2"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+	"golang.org/x/sys/unix"
+)
+
+const volumeIDXattrKey = "trusted.glusterfs.volume-id"
+
+func snapRestore(c transaction.TxnCtx) error {
```

Comment:
Any solution to keep the same brick path as original brick? Unmount original brick and mount Snapbrick to original path?

Reply:
yes.

Reply:
Please ignore the previos command.
When we restore, it is not sure the whther the old directory is exist or not. Let's say we had done a remove brick after taking snapshot, then we cannot assure about the existance. In this case it is also possible that the directory might have used for some other mount or as a non-gluster mount.